src/main/java/org/incodelearning/algorithm/SinglyLinkedList.java

```diff
@@ -0,0 +1,129 @@
+package org.incodelearning.algorithm;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * <p>Default linked list implementation, singly linked. Providing access to the first element only.
+ * <p>All operations take constant time except iterator O(n).
+ */
+public class SinglyLinkedList<E> extends AbstractLinkedList<E> implements Stack<E> {
+
+    protected int N;
+    protected Node<E> first;
+
+    @Override
+    public Iterator<E> iterator() {
+        return new LinkedListIterator<>(first);
+    }
+
+    private class LinkedListIterator<E> implements Iterator<E> {
+        private Node<E> current;
+
+        /**
+         * Does not compile without constructor argument, type mismatch.
+         */
+        public LinkedListIterator(Node<E> first) {
+            current = first;
+        }
+
+        @Override
+        public boolean hasNext() {
+            return current != null;
+        }
+
+        @Override
+        public E next() {
+            if (!hasNext()) throw new NoSuchElementException();
+            E element = current.element;
+            current = current.next;
+            return element;
+        }
+    }
+
+    protected static class Node<E> {
+        protected E element;
+        protected Node<E> next;
+    }
+
+    /**
+     * Initializes an empty singly linked list.
+     */
+    public SinglyLinkedList() {
+        N = 0;
+        first = null;
+    }
+
+    /**
+     * Pushes the element on top of the stack represented by this singly linked list.
+     *
+     * @param element the element to add
+     */
+    public void push(E element) {
```

Comment:
Based on your implementation of Iterator, null is not allowed in your list. Thus, here you would need to check whether element is null, if so, throw an exception.